dsls/sql/src/test/java/org/apache/beam/dsls/sql/interpreter/BeamSQLFnExecutorTest.java

```diff
@@ -119,6 +121,37 @@ public void testBeamProjectRel() {
 
 
   @Test
+  public void testBuildExpression_logical() {
```

Comment:
I see a lot of tests verifying things work when given correct input. I generally feel better when I see tests validating behaviour is correct when given incorrect input as well. Could you add some additional tests exercising failure paths as well?